components/tests/python/library/__init__.py

```diff
@@ -747,9 +747,14 @@ def loginAttempt(self, name, t, pw="BAD", less=False):
     def doSubmit(self, request, client, test_should_pass=True,
                  omero_group=None):
         """
-        Performs the request waits on completion and checks that the
-        result is not an error.
+        Performs the request(s), waits on completion and checks that the
+        result is not an error. The request can either be a single command
+        or a list of commands. If the latter then the request list will be
+        wrapped in a DoAll.
         """
+        if isinstance(request, list):
```

Comment:
and tuples a la [iter()](http://stackoverflow.com/questions/1952464/in-python-how-do-i-determine-if-an-object-is-iterable)?